`AssertionError: (Base.JLOptions()).depwarn == 1` when we run the Pkg tests with `Base.runtests(["Pkg"])`

```
ERROR: LoadError: AssertionError: (Base.JLOptions()).depwarn == 1
Stacktrace:
 [1] top-level scope
   @ /tmp/jl_kJ4snX/TestArguments/test/runtests.jl:4
 [2] include(fname::String)
   @ Base.MainInclude ./client.jl:451
 [3] top-level scope
   @ none:6
in expression starting at /tmp/jl_kJ4snX/TestArguments/test/runtests.jl:4
```

The full Buildkite log is here: https://buildkite.com/julialang/julia-master/builds/5345#6729df4a-5129-4510-a957-940bc1ed473c

This is a Buildkite job triggered on the following pull request: https://github.com/JuliaLang/julia/pull/42885

--- a/test/test_packages/TestArguments/test/runtests.jl
+++ b/test/test_packages/TestArguments/test/runtests.jl
@@ -1,4 +1,4 @@
 @assert ARGS == ["a", "b"]
 @assert Base.JLOptions().quiet == 1 # --quiet
 @assert Base.JLOptions().check_bounds == 2 # overriden default when testing
-@assert Base.JLOptions().depwarn == 1 # Default to depwarn on
+@assert (Base.JLOptions().depwarn == 1) || (Base.JLOptions().depwarn == 2)

